### PR TITLE
Add Project API

### PR DESCRIPTION
**Pull request by rafael-azevedo**
This PR adds the project API.  To do so I moved common types between the project and projectClaim types to a common files. I also added updated examples of both types.

### CODE REVIEW

**Comment by rafael-azevedo** [comment]
When testing this implementation I realized that Kube has a default type called projects. While this implementation does work you can get the object by saying `oc get projects.gcp.managed.openshift.io` we many want to rethink the name of this api for more clarity .

**Comment by NautiluX** [comment]
I'm in favor of renaming. GCPProject maybe?

**Comment by jhernand** [comment]
@tzvatot please review.

File: deploy/crds/gcp_v1alpha1_project_crd.yaml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  - JSONPath: .status.state
+    description: Status of the project
+    name: State
+    type: string
+  - JSONPath: .spec.projectClaimCRLink.name
+    description: Name of corresponding project claim CR
+    name: ClaimName
+    type: string
+  - JSONPath: .spec.projectClaimCRLink.namespace
+    description: Namesspace of corresponding project CR
```
Comment (NautiluX) [inline deploy/crds/gcp_v1alpha1_project_crd.yaml]: should this be project claim CR?

File: pkg/apis/gcp/v1alpha1/common_types.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package v1alpha1
+
+// LegalEntity contains Red Hat specific identifiers to the original creator the clusters
+type LegalEntity struct {
+	Name string `json:"name"`
+	ID   string `json:"id"`
+}
+
+// NamespacedName contains the name of a secret and its namespace
```
Comment (NautiluX) [inline pkg/apis/gcp/v1alpha1/common_types.go]: We use this as generic name of any Kubernetes object, right? Like, as a reference to a `ProjectClaim`, so I assume it's not just secret.

File: pkg/apis/gcp/v1alpha1/project_types.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
```
Comment (NautiluX) [inline pkg/apis/gcp/v1alpha1/project_types.go]: Shouldn't we remove this comment?

Comment (rafael-azevedo) [reply pkg/apis/gcp/v1alpha1/project_types.go]: We have left this in on other repos, but I don't see why we shouldn't remove it.  Do you think we should just remove it ?

**Comment by rafael-azevedo** [comment]
Updated the type to name `GCPProject` and changed the `GCPProjectName` field in both api types to `GCPProjectID` for better ease of reading. The `GCPProjectID` is the name of the Project have have instantiated in the GCP cloud and not the CR name.

File: pkg/apis/gcp/v1alpha1/project_types.go (hunk 3 above)
Comment (NautiluX) [reply pkg/apis/gcp/v1alpha1/project_types.go]: I understand it as note by framework for us to remove once we understood this file is for us to touch ;)

Comment (rafael-azevedo) [reply pkg/apis/gcp/v1alpha1/project_types.go]: I will remove on all files in this operator

**Pull request merged**